scripts/workers/code_upload_submission_worker.py

```diff
@@ -550,38 +552,60 @@ def update_failed_jobs_and_send_logs(
     clean_submission = False
     try:
         job_def = read_job(api_instance, job_name)
-        controller_uid = job_def.metadata.labels["controller-uid"]
-        pod_label_selector = "controller-uid=" + controller_uid
-        pods_list = core_v1_api_instance.list_namespaced_pod(
-            namespace="default",
-            label_selector=pod_label_selector,
-            timeout_seconds=10,
-        )
-        container_state_map = {}
-        for container in pods_list.items[0].status.container_statuses:
-            container_state_map[container.name] = container.state
-        for container_name, container_state in container_state_map.items():
-            if container_name in ["agent", "submission"]:
-                if container_state.terminated is not None:
-                    if container_state.terminated.reason == "Error":
-                        pod_name = pods_list.items[0].metadata.name
-                        try:
-                            pod_log_response = (
-                                core_v1_api_instance.read_namespaced_pod_log(
-                                    name=pod_name,
-                                    namespace="default",
-                                    _return_http_data_only=True,
-                                    _preload_content=False,
-                                    container=container_name,
-                                )
-                            )
-                            pod_log = pod_log_response.data.decode("utf-8")
-                            clean_submission = True
-                            submission_error = pod_log
-                        except client.rest.ApiException as e:
-                            logger.exception(
-                                "Exception while reading Job logs {}".format(e)
-                            )
+        if job_def:
```

Comment:
@savish28 `read_job` will return a valid kubernetes job if the job is in pending state. Have you tested this change?

Reply:
@Ram81 this is the case when I manually delete the job from kubectl and in this case read_job would return None, this is a different usecase.

Reply:
Yeah, I have tested this change

Reply:
Is this the reason why fargate worker was failing?

Reply:
Not sure, why the fargate was restarting is still a mystery